src/main.ts

```diff
@@ -18,11 +18,12 @@ import {SocketClientState} from './common';
 
 class RubyDebugSession extends DebugSession {
 
-	// we don't support multiple threads, so we can use a hardcoded ID for the default thread
-	private static THREAD_ID = 2;
-
 	private _breakpointId = 1000;
+	private _threadId = 2;
+	private _frameId = 0;
+	private _hasStopped = false;
```

Comment:
`_hasStopped` is if `.Run('start')` has been called. It enables display of `stopped on entry` instead of a standard step, which doesn't show anything. So it's not the state of the socket (which `.state` tracks), rather whether or not a suspended has already been received.